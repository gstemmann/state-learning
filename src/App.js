// import logo from './logo.svg';
// import Circle from "./Circle";
// import ColoredCircles from "./ColoredCircles";
// // import Button from './button';
// import Counter from './Counter';
// import ShoppingList from './ShoppingList';
// // import Group from './Group';
// import NumbersList from './NumbersList';
// import UserForm from './UserForm';
// import ShoppingList from './ShoppingList';
// import Toggler from './Toggler';
// import SimpleForm from './SimpleForm';
// import ProfileViewer from './ProfileViewer';
// import ProfileSearchForm from './ProfileSearchForm';
// import ProfileViewerWithSearch from './ProfileViewerWithSearch';
import Timer from './Timer';



function App() {
  return (
    <div className="App">
      <Timer />
    </div>
  );
}

export default App;
