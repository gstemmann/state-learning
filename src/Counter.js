import React, { useState } from "react";



const Counter = () => {
    const [currentState, changeState] = useState(1);
    const clickUp = () => changeState(currentState + 1);
    const clickUpBy2 = () => changeState(currentState + 2);
    
    return (
        <>
        <h1> Counter is at : {currentState}</h1>
        <button onClick={clickUp}>+</button>
        <button onClick={clickUpBy2}>+2</button>
        </>

    )
}

export default Counter;