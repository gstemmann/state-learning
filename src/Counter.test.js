import React from "react";
import { render, screen } from '@testing-library/react';
import App from './App';
import Counter from './Counter';



it('renders the Counter component', () => {
  render(<Counter />);
});

// snapshot test
it("matches snapshot", function() {
    const {asFragment} = render(<Counter />);
    expect(asFragment()).toMatchSnapshot();
  });