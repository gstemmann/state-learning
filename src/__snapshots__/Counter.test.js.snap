// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h1>
     Counter is at : 1
  </h1>
  <button>
    +
  </button>
  <button>
    +2
  </button>
</DocumentFragment>
`;
